pkg/controller/syncmachineset/syncmachineset_controller_test.go

```diff
@@ -0,0 +1,475 @@
+				assert.Nil(t, createdSS.syncSet, "no call expected to created/update the syncset when paused")
+			},
+			skipGenerateCall: true,
+		},
+		{
+			name: "error on generate error",
+			existingObjects: []runtime.Object{
+				testClusterDeployment(),
+				testClusterSecret(),
+			},
+			generateMSResponse: generateMachineSetsResponse{
+				err: fmt.Errorf("TEST ERROR"),
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClient(test.existingObjects...)
+
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			mockActuator := mockactuator.NewMockActuator(mockCtrl)
+
+			if !test.skipGenerateCall {
+				mockActuator.EXPECT().GenerateMachineSets(gomock.Any(), gomock.Any(), gomock.Any()).Return(
+					test.generateMSResponse.machineSets, test.generateMSResponse.err,
+				)
+			}
+
+			applier := &fakeApplier{}
+
+			rcd := &ReconcileSyncMachineSet{
+				Client: fakeClient,
+				scheme: scheme.Scheme,
+				logger: log.WithField("controller", "syncmachineset"),
+				actuatorBuilder: func(client.Client, *hivev1.ClusterDeployment, log.FieldLogger) (Actuator, error) {
+					return mockActuator, nil
+				},
+				kubeCLI: applier,
+			}
+
+			_, err := rcd.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      testName,
+					Namespace: testNamespace,
+				},
+			})
+
+			if test.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				if test.validate != nil {
+					test.validate(t, fakeClient, test.expectedMachineSets, applier.createdSyncSet)
+				} else {
+					validateSyncSet(t, test.expectedMachineSets, applier.createdSyncSet)
+				}
+			}
+		})
+	}
+}
+
+func validateSyncSet(t *testing.T, expectedMS []createdMachineSetInfo, createdSS createdSyncSetInfo) {
+	assert.Equal(t, len(expectedMS), len(createdSS.resources), "unexpected number of machineset entries in syncset")
+	for _, expected := range expectedMS {
+		found := false
+
+		for _, res := range createdSS.resources {
+			if res.name == expected.name {
+				found = true
+				assert.Equal(t, expected.replicas, res.replicas, "mismatched replicas")
+				assert.Equal(t, expected.instanceType, res.instanceType, "mismatched instance type")
+			}
+		}
+		assert.True(t, found, "did not find machineset %s in syncset", expected.name)
+	}
+}
+
+func testMachineSet(name string, replicas int, machineType string) *machineapi.MachineSet {
+	ms := &machineapi.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: machineapi.MachineSetSpec{
+			Replicas: pointer.Int32Ptr(int32(replicas)),
+			Template: machineapi.MachineTemplateSpec{
+				Spec: machineapi.MachineSpec{
+					ProviderSpec: machineapi.ProviderSpec{
+						Value: &runtime.RawExtension{Object: &gcpprovider.GCPMachineProviderSpec{
+							MachineType: machineType,
+						},
+						},
```

Comment:
Combine this onto the previous line since the corresponding opening braces are on the same line.